lib/livechat/index.js

```diff
@@ -63,6 +78,34 @@ LiveChat.prototype.loaded = function(){
 };
 
 /**
+ * Listen for chat events.
+ */
+
+LiveChat.prototype.attachListeners = function(){
+  var self = this;
+  window.LC_API = window.LC_API || {};
+  window.LC_API.on_chat_started = function(data){
+    self.analytics.track('Live Chat Conversation Started',
+      { agentName: data.agent_name },
+      { context: { integration: integration }});
+  };
+  window.LC_API.on_message = function(data){
+    if (data.user_type === 'visitor') {
+      self.analytics.track('Live Chat Message Sent',
+        { messageBody: data.text },
+        { context: { integration: integration }});
+    } else {
```

Comment:
If Livechat defines `data.user_type` as e.g. `agent`, I'd refactor this as a separate `if` block:

```javascript
if (data.user_type === 'agent') {
  // ...
}
```

That way if they change their API to include a new different `user_type` without notification, we don't inadvertently send those along as received messages.